src/iac-mcp-server/awslabs/iac_mcp_server/deployment_troubleshooter.py

```diff
@@ -0,0 +1,227 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta, timezone
+from typing import Dict, List, Any, Optional
+from dateutil import parser as dateutil_parser
+import json
+import boto3
+import botocore.config
+
+# CloudFormation's source IP in CloudTrail events
+CLOUDTRAIL_SOURCE_IP_FOR_CLOUDFORMATION = 'cloudformation.amazonaws.com'
+
+session_config = botocore.config.Config(
+    user_agent_extra='iac-mcp-server/1.0.0',
+)
+
+
+class DeploymentTroubleshooter:
```

Comment:
Do we plan to add tests for this as well? It's possible I am missing something, but I don't see tests for this in the PR.